.github/workflows/mobile.yml

```diff
@@ -0,0 +1,97 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+
+name: Mobile CI
+
+on:
+  pull_request:
+    types: [ready_for_review, opened, reopened, synchronize]
+    paths:
+      - '.github/workflows/mobile.yml'
+      - 'mobile/**'
+
+env:
+  SERVICE_NAME: 'mobile'
+  NODEJS_VERSION: '16.17.1'
+  PNPM_VERSION: '8.3.1'
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+
+    steps:
+      - name: Checkout Codebase
+        uses: actions/checkout@v3
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build and push image
+        uses: docker/build-push-action@v3
+        with:
+          context: ./${{ env.SERVICE_NAME }}
+          file: ./${{ env.SERVICE_NAME }}/Dockerfile
+          push: true
+          tags: ${{ secrets.DOCKER_HUB_USERNAME }}/codenaute:mobile
+
+  workflow:
+    needs: build-and-push
+    runs-on: ubuntu-latest
+    if: github.event.pull_request.draft == false
+
+    container:
+      image: 'reynandev/codenaute:mobile'
+      env:
+        NODE_ENV: test
+      ports:
+        - 3000:3000
+      volumes:
+        - ${{ github.workspace }}:/app
+
+    steps:
+      - name: Checkout Codebase
+        uses: actions/checkout@v3
+
+      - name: Install dependencies globaly
+        run: pnpm install --fix-lockfile
+
+      - name: Install dependencies on client
+        run: pnpm install --fix-lockfile
+        working-directory: ./${{ env.SERVICE_NAME }}
+
+      - name: Restore Github Cache
+        uses: actions/cache@v3
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.OS }}-platform-client-${{ hashFiles('pnpm-lock.yaml', 'client/pnpm-lock.yaml') }}
+
+      - name: Run Tests
+        run: pnpm test:ci
+        working-directory: ./${{ env.SERVICE_NAME }}
+
+  # build:
```

Comment:
Are you sur we will need this part? I think we should prevent comments :)

Reply:
You are right, removed comments.